### PR TITLE
feat: Refactor and improve our create-module script

### PR DESCRIPTION
**Pull request by anicholls**
## Summary

Our `create-module` script started a simple shell script to create some files from a template. Over time this ballooned and conditionals were added. Needless to say, it got quite messy as a shell script. This PR refactors it into a nice little node CLI.

This also includes some improvements like the ability to pick which targets you would like to generate (previously React was required), and more helpful templates.

Closes https://github.com/Workday/canvas-kit/issues/132

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] `yarn test` passes

## Additional References
image (23)

Scalable targets:
image (24)

### CODE REVIEW

File: utils/create-component/templates/css/packageJson.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+module.exports = (name, description) => `
+{
+	"name": "@workday/canvas-kit-css-${name}",
```
Comment (anicholls) [inline utils/create-component/templates/css/packageJson.js]: Need to change these to spaces.

File: utils/create-component/templates/css/packageJson.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "files": [
+    "dist",
+    "lib",
+    "index.scss"
+  ],
+  "style": "dist/canvas-kit-css-${name}.min.css",
+  "main": "dist/canvas-kit-css-${name}.min.css",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/Workday/canvas-kit/tree/master/modules/${name}/css"
```
Comment (sahlhoff) [inline utils/create-component/templates/css/packageJson.js]: We'll need `/modules/_labs/` here for the unstable components.

File: utils/create-component/templates/css/packageJson.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	},
+	"scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (sahlhoff) [inline utils/create-component/templates/css/packageJson.js]: For some reason the escape characters here are missing in the `package.json` that gets written: 

`"test": "echo "Error: no test specified" && exit 1",`

File: utils/create-component/templates/css/packageJson.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "build": "node ../../../utils/css-build.js index.scss"
```
Comment (sahlhoff) [inline utils/create-component/templates/css/packageJson.js]: We'll have to add an extra `../` for the `_labs` components.

File: utils/create-component/templates/css/packageJson.js (hunk 1 above)
Comment (sahlhoff) [reply utils/create-component/templates/css/packageJson.js]: Also need to add `canvas-kit-<labs>-css` here for unstable components

**Pull request merged**